### ISSUE TITLE
Don't constrain the 3D editor camera to a pitch angle of -90..90 degrees

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot-proposals#1371]
*Note: I'm not 100% convinced doing this is a good idea. This is why I'm opening a proposal about it here, to see whether this behavior should actually be changed or left as-is.*

**Describe the project you are working on:**

The Godot editor :slightly_smiling_face: 

**Describe the problem or limitation you are having in your project:**

Many 3D editors such as Blender don't constrain the pitch angle when orbiting or using freelook mode. I wonder if the Godot editor should do the same.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**

We can change the editor code to remove the 3D editor camera's pitch angle constraint.

*Advantages:*

- Selecting 3D nodes that are above/below you becomes easier.
- Previewing 3D scenes for correct appearance becomes easier.
- This makes the camera behavior more consistent with Blender.

*Disadvantages:*

- It's easier to get disoriented since the camera can end up upside-down.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**

Remove [this line](https://github.com/godotengine/godot/blob/4ee779e6d307d60c8f38c0927f0e951b95eb0481/editor/plugins/node_3d_editor_plugin.cpp#L2141) and its equivalent in the pan code. Make sure input works as expected when the camera is upside down – the X axis should probably be inverted automatically when the pitch angle isn't between -90 and 90 degrees.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

No, as this is core editor functionality.

**Is there a reason why this should be core and not an add-on in the asset library?:**

This is core editor functionality.

**Comment by RaTcHeT302** [comment]
i feel like, we shouldn't allow the camera to go upside down, but you should be able to look at the ground straight down

i would allow the camera to pitch at any angle, but i just wouldn't allow it to go upside down as it would get freaky
it should be pretty easy to tell when, you are looking straight down, or straight up, so it should be easy to fix

**Comment by Zireael07** [comment]
@RaTcHeT302: Nope, it's not easy to tell when we're upside down due to something called gimbal lock. We'd have to use quaternions...
Other than that, your idea is very good.

**Comment by RobertBColton** [comment]
The editor should be using Quaternions like Blender. Blender does not have any limitation to where you can look and the camera does not flip upside down. Quaternion math will invert the up vector so the camera doesn't appear to flip when you cross the -90 degree threshold.

Don't think about it, just go try Blender and see that it can be done seamlessly.

**Comment by Calinou** [comment]
> and the camera does not flip upside down

It still does as of Blender 4.3.1:

https://github.com/user-attachments/assets/bde83f39-84a3-426c-88e1-4164c1662b1e

Notice how the horizontal rotation is inverted until I release the mouse and click it again, since it's only inverted if the camera is upside down. However, this check is performed when you *start* dragging the mouse - it's not performed continuously.

**Comment by RobertBColton** [comment]
Ok, that's interesting. The alternative is you can flip it over once you cross -90 though.

I'm not sure which way I think is best for Godot.

I do think if we go with Blenders way, we should consider a second gizmo compass 🧭pointing at the up vector to make it easy to tell at a glance if you're inverted.
